Jenkinsfile

```diff
@@ -0,0 +1,121 @@
+@Library('github.com/connexta/cx-pipeline-library@master') _
+@Library('github.com/connexta/github-utils-shared-library@master') __
+
+pipeline {
+    agent {
+        node {
+            label 'linux-small'
+            customWorkspace "/jenkins/workspace/${JOB_NAME}/${BUILD_NUMBER}"
+        }
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr:'25'))
+        disableConcurrentBuilds()
+        timestamps()
+        skipDefaultCheckout()
+    }
+    triggers {
+        /*
+          Restrict nightly builds to master branch, all others will be built on change only.
+          Note: The BRANCH_NAME will only work with a multi-branch job using the github-branch-source
+        */
+        cron(BRANCH_NAME == "master" ? "@weekly" : "")
+		pollSCM("H/10 * * * *")
+    }
+    environment {
+        LARGE_MVN_OPTS = '-Xmx8192M -Xss128M -XX:+CMSClassUnloadingEnabled -XX:+UseConcMarkSweepGC '
+        LINUX_MVN_RANDOM = '-Djava.security.egd=file:/dev/./urandom'
+        GITHUB_USERNAME = 'codice'
+        GITHUB_TOKEN = credentials('github-api-cred')
+        GITHUB_REPONAME = 'usng4j'
+    }
+    stages {
+        stage('Setup') {
+            steps {
+                dockerd {}
+                slackSend color: 'good', message: "STARTED: ${JOB_NAME} ${BUILD_NUMBER} ${BUILD_URL}"
+                postCommentIfPR("Internal build has been started, your results will be available at build completion.", "${GITHUB_USERNAME}", "${GITHUB_REPONAME}", "${GITHUB_TOKEN}")
+            }
+        }
+
+        // Checkout the repository
+        stage('Checkout repo') {
+            steps {
+                retry(3) {
+                    checkout([$class: 'GitSCM', branches: [[name: "refs/heads/${BRANCH_NAME}"]], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'WipeWorkspace'], [$class: 'LocalBranch', localBranch: "${BRANCH_NAME}"], [$class: 'CloneOption', timeout: 30, shallow: true]], submoduleCfg: [], userRemoteConfigs: scm.userRemoteConfigs ])
+                }
+            }
+        }
+
+        stage('Full Build except itest') {
```

Comment:
I'm not sure we need this stage at all in this repo. There are no itests.

Reply:
Did I categorize this stage correctly, based on the flags set? I can easily remove this stage.

Reply:
Removed stage